test/rockpaperscissors.js

```diff
@@ -0,0 +1,772 @@
+
+    beforeEach("Deploy and prepare", async function() {
+        rcp = await Game.new({from: contractOwner});
+        gameKey = await rcp.createPlayerOneMoveHash(playerOne, playerOneSecretBytes32, ROCK);
+        playerTwoMoveHash = await rcp.createPlayerTwoMoveHash(playerTwo, gameKey, playerTwoSecretBytes32, PAPER);
+        const snapShot = await timeMachine.takeSnapshot();
+        snapshotId = snapShot['result'];
+    });
+
+    afterEach(async() => {
+        await timeMachine.revertToSnapshot(snapshotId);
+    });
+
+    describe("Player moves", async () => {
+
+          it("Player one hashes their address, secret and move", async () => {
+                gameKey = await rcp.createPlayerOneMoveHash(playerOne, playerOneSecretBytes32, ROCK);
+                const expectedMoveHash =  await soliditySha3(playerOne, playerOneSecretBytes32, ROCK);
+                assert.strictEqual(expectedMoveHash, gameKey);
+            });
+
+            it("Player one creates a game", async () => {
+                const txObj = await rcp.movePlayerOne(gameKey, toBN(10), {from: playerOne, value: toBN(10)});
+
+               
+
+                await truffleAssert.eventEmitted(txObj, 'PlayerOneMoves', (ev) => {
+
+                    return  ev.player === playerOne &&
+                            ev.gameKey === gameKey &&
+                            ev.stake.toString(10) === "10" &&
+                            ev.amount.toString(10) === "10"
+                },'PlayerOneMoves event is emitted');
+
+                const game = await rcp.games(gameKey);
+
+                assert.strictEqual(game.stake.toString(10), "10");
+                assert.strictEqual(game.playerOne, playerOne);
+                assert.strictEqual(game.step.toString(10), PLAYER_ONE_MOVE);
+
+                const playerOneOwed = toBN(await rcp.balances(playerOne));
+
+                assert.strictEqual(playerOneOwed.toString(10), "0");
+            });
+
+        it("Value sent that exceeds the stake, is sent to the player's balance", async () => {
+            gameKey = await rcp.createPlayerOneMoveHash(playerOne, playerOneSecretBytes32, PAPER);
+
+            const initPlayerOneOwed = await rcp.balances(playerOne);
+            assert.strictEqual(initPlayerOneOwed.toString(10), "0");
+
+            // 7 - 5 = 2 is added to the balance
+            await rcp.movePlayerOne(gameKey, toBN(5), {from: playerOne, value: toBN(7)});
+
+            const playerOneOwed1 = await rcp.balances(playerOne);
+            assert.strictEqual(playerOneOwed1.toString(10), "2");
+
+            // 10 - 5 = 5 is added to the balance
+            gameKey = await rcp.createPlayerOneMoveHash(playerOne, playerOneSecretBytes32, ROCK);
+            await rcp.movePlayerOne(gameKey, toBN(5), {from: playerOne, value: toBN(10)});
+
+            // Total balance is now 5 + 2
+            const playerOneOwed2 = await rcp.balances(playerOne);
+            assert.strictEqual(playerOneOwed2.toString(10), "7");
```

Comment:
Do you need the `.toString(10)`?

Reply:
I get:

`AssertionError: expected <BN: 7> to equal 7`

Reply:
Oh, then on the other lines, you don't need to do `toBN(playerOneOwed2)`